client/rpc/src/eth.rs

```diff
@@ -1937,3 +1972,74 @@ where
 		}
 	}
 }
+
+/// Stores an LRU cache for block data and their transaction statuses.
+/// These are large and take a lot of time to fetch from the database.
+/// Storing them in an LRU cache will allow to reduce database accesses
+/// when many subsequent requests are related to the same blocks.
+pub struct EthBlockDataCache<B: BlockT> {
+	blocks: Mutex<LruCache<B::Hash, EthereumBlock>>,
```

Comment:
Please change all `Mutex` usage in this file to `parking_lot::Mutex`. Might also require you to change some unrelated code.

Reply:
Also, isn't it better if we use `RwLock` here?

Reply:
I see a few other std `Mutex`. If parking_lot is better, it would be a good follow-up PR to migrate all `Mutex` to parking_lot then ? I'll change only those new `Mutex` for the parking_lot version in this PR.

Reply:
Can you also change it to RwLock? Or what are the reasons not to?

Reply:
Since it's an LRU Cache `get` also take an `&mut self` since it'll bump the entry to the top of the cache.